src/utils/delegationUtils.ts

```diff
@@ -1,44 +1,44 @@
 import {
-  createCaveatBuilder,
   createDelegation,
   createExecution,
   Delegation,
   MetaMaskSmartAccount,
-} from "@metamask/delegation-toolkit";
-import { DelegationManager } from "@metamask/delegation-toolkit/contracts";
-import { SINGLE_DEFAULT_MODE } from "@metamask/delegation-toolkit/utils";
-import { Address, Hex, zeroAddress } from "viem";
+  ExecutionMode,
+} from "@metamask/smart-accounts-kit";
+import { DelegationManager } from "@metamask/smart-accounts-kit/contracts";
+import { Address, Hex, parseEther, zeroAddress } from "viem";
 
 export function prepareRootDelegation(
   delegator: MetaMaskSmartAccount,
   delegate: Address
 ): Delegation {
-  // The following caveat enforcer is a simple example that limits
-  // the number of executions the delegate can perform on the delegator's
+  // The following scope is a simple example that limits
+  // the number of native token transfers the delegate can perform on the delegator's
```

Comment:
"... that limits the amount of native token that the delegate can transfer ..." 

as it stands, it feels like it's just a call limit.

Reply:
Nice catch!